Tools/cases_generator/lexer.py

```diff
@@ -13,78 +15,74 @@ def choice(*opts: str) -> str:
 
 # Regexes
 
-# Longer operators must go before shorter ones.
-
-PLUSPLUS = r"\+\+"
-MINUSMINUS = r"--"
-
-# ->
-ARROW = r"->"
-ELLIPSIS = r"\.\.\."
-
-# Assignment operators
-TIMESEQUAL = r"\*="
-DIVEQUAL = r"/="
-MODEQUAL = r"%="
-PLUSEQUAL = r"\+="
-MINUSEQUAL = r"-="
-LSHIFTEQUAL = r"<<="
-RSHIFTEQUAL = r">>="
-ANDEQUAL = r"&="
-OREQUAL = r"\|="
-XOREQUAL = r"\^="
-
-# Operators
-PLUS = r"\+"
-MINUS = r"-"
-TIMES = r"\*"
-DIVIDE = r"/"
-MOD = r"%"
-NOT = r"~"
-XOR = r"\^"
-LOR = r"\|\|"
-LAND = r"&&"
-LSHIFT = r"<<"
-RSHIFT = r">>"
-LE = r"<="
-GE = r">="
-EQ = r"=="
-NE = r"!="
-LT = r"<"
-GT = r">"
-LNOT = r"!"
-OR = r"\|"
-AND = r"&"
-EQUALS = r"="
-
-# ?
-CONDOP = r"\?"
-
-# Delimiters
-LPAREN = r"\("
-RPAREN = r"\)"
-LBRACKET = r"\["
-RBRACKET = r"\]"
-LBRACE = r"\{"
-RBRACE = r"\}"
-COMMA = r","
-PERIOD = r"\."
-SEMI = r";"
-COLON = r":"
-BACKSLASH = r"\\"
-
-operators = {op: pattern for op, pattern in globals().items() if op == op.upper()}
-for op in operators:
-    globals()[op] = op
-opmap = {pattern.replace("\\", "") or "\\": op for op, pattern in operators.items()}
+# Mapping from operator names to their regular expressions.
+operators = {
+    # Longer operators must go before shorter ones.
+    (PLUSPLUS := "PLUSPLUS"): r'\+\+',
+    (MINUSMINUS := "MINUSMINUS"): r"--",
+    # ->
+    (ARROW := "ARROW"): r"->",
+    (ELLIPSIS := "ELLIPSIS"): r"\.\.\.",
+    # Assignment operators
+    (TIMESEQUAL := "TIMESEQUAL"): r"\*=",
+    (DIVEQUAL := "DIVEQUAL"): r"/=",
+    (MODEQUAL := "MODEQUAL"): r"%=",
+    (PLUSEQUAL := "PLUSEQUAL"): r"\+=",
+    (MINUSEQUAL := "MINUSEQUAL"): r"-=",
+    (LSHIFTEQUAL := "LSHIFTEQUAL"): r"<<=",
+    (RSHIFTEQUAL := "RSHIFTEQUAL"): r">>=",
+    (ANDEQUAL := "ANDEQUAL"): r"&=",
+    (OREQUAL := "OREQUAL"): r"\|=",
+    (XOREQUAL := "XOREQUAL"): r"\^=",
+    # Operators
+    (PLUS := "PLUS"): r"\+",
+    (MINUS := "MINUS"): r"-",
+    (TIMES := "TIMES"): r"\*",
+    (DIVIDE := "DIVIDE"): r"/",
+    (MOD := "MOD"): r"%",
+    (NOT := "NOT"): r"~",
+    (XOR := "XOR"): r"\^",
+    (LOR := "LOR"): r"\|\|",
+    (LAND := "LAND"): r"&&",
+    (LSHIFT := "LSHIFT"): r"<<",
+    (RSHIFT := "RSHIFT"): r">>",
+    (LE := "LE"): r"<=",
+    (GE := "GE"): r">=",
+    (EQ := "EQ"): r"==",
+    (NE := "NE"): r"!=",
+    (LT := "LT"): r"<",
+    (GT := "GT"): r">",
+    (LNOT := "LNOT"): r"!",
+    (OR := "OR"): r"\|",
+    (AND := "AND"): r"&",
+    (EQUALS := "EQUALS"): r"=",
+    # ?
+    (CONDOP := "CONDOP"): r"\?",
+    # Delimiters
+    (LPAREN := "LPAREN"): r"\(",
+    (RPAREN := "RPAREN"): r"\)",
+    (LBRACKET := "LBRACKET"): r"\[",
+    (RBRACKET := "RBRACKET"): r"\]",
+    (LBRACE := "LBRACE"): r"\{",
+    (RBRACE := "RBRACE"): r"\}",
+    (COMMA := "COMMA"): r",",
+    (PERIOD := "PERIOD"): r"\.",
+    (SEMI := "SEMI"): r";",
+    (COLON := "COLON"): r":",
+    (BACKSLASH := "BACKSLASH"): r"\\",
+}
+__all__.extend(operators.keys())
+opmap = {__pattern.replace("\\", "") or "\\": __opname
```

Comment:
You can use regular names and just `del` them later.

Reply:
Oh yes, that's maybe easier. I tend to forget about this pattern :) thanks!

Reply:
Err.. I think we both assumed that those names are actually polluting the global namespace but that's not the case :D it's only if we were to use loops!